pkg/cmd/infra/router/template_test.go

```diff
@@ -0,0 +1,132 @@
+package router
+
+import (
+	"fmt"
+	"github.com/google/go-cmp/cmp"
+	templateplugin "github.com/openshift/router/pkg/router/template"
+	"strings"
+	"testing"
+)
+
+func TestParseHeadersToBeSetOrDeleted(t *testing.T) {
+	testCases := []struct {
+		description        string
+		inputValue         string
+		expectedValue      []templateplugin.HTTPHeader
+		expectError        bool
+		expectErrorMessage string
+	}{
+		{
+			description: "should percent decode the single header value",
+			inputValue:  "Accept:text%2Fplain%2C+text%2Fhtml:Set",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "Accept", Value: "'text/plain, text/html'", Action: "Set"},
+			},
+			expectError: false,
+		},
+		{
+			description:        "should not percent decode the invalid double encoded single header value",
+			inputValue:         "Content-Location%3A%252Fmy-first-blog-post%3ASet",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header input specification",
+		},
+		{
+			description: "should percent decode the multiple response header values",
+			inputValue:  "X-Frame-Options:DENY:Set,X-XSS-Protection:1%3Bmode%3Dblock:Set,x-forwarded-client-cert:%25%7B%2BQ%7D%5Bssl_c_der%2Cbase64%5D:Set",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "X-Frame-Options", Value: "'DENY'", Action: "Set"},
+				{Name: "X-XSS-Protection", Value: "'1;mode=block'", Action: "Set"},
+				{Name: "x-forwarded-client-cert", Value: "'%{+Q}[ssl_c_der,base64]'", Action: "Set"},
+			},
+			expectError: false,
+		},
+		{
+			description: "should percent decode the multiple request header values",
+			inputValue:  "Accept:text%2Fplain%2C+text%2Fhtml:Set,Accept-Encoding:Delete",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "Accept", Value: "'text/plain, text/html'", Action: "Set"},
+				{Name: "Accept-Encoding", Action: "Delete"},
+			},
+			expectError: false,
+		},
+		{
+			description: "should percent decode the multiple header values with simple non encoded strings",
+			inputValue:  "X-Frame-Options:DENY:Set",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "X-Frame-Options", Value: "'DENY'", Action: "Set"},
+			},
+			expectError: false,
+		},
+		{
+			description:        "When Action is `Set` it should not percent decode the incorrectly encoded value part Eg: percent is not followed by two hexadecimal digits",
```

Comment:
```suggestion
			description:        "when Action is `Set` it should report an error on incorrectly encoded value part (percent is not followed by two hexadecimal digits)",
```

Reply:
Lower case first letter on the test case descriptions, everywhere, please.

Reply:
ok